source/MaterialXGenShader/Util.cpp

```diff
@@ -951,10 +951,79 @@ void getUdimScaleAndOffset(const vector<Vector2>& udimCoordinates, Vector2& scal
     offsetUV[1] = -minUV[1];
 }
 
-bool connectsToNormalMapNode(OutputPtr output)
+NodePtr connectsToNodesOfCategory(OutputPtr output, const StringSet& categories)
```

Comment:
This might be more clear as "connectsToNodeOfCategory", since only a single connected node is returned.